### ISSUE TITLE
Component decomposition

### ISSUE THREAD

**Issue by vijoc** [alikucukavci/FSO#24]
There is a problem that at least I keep running into every now and then.

Currently, when modeling heat exchangers, the simple choice is to model them as a single component of class `fso:EnergyConversionDevice`. This is fine in itself, but leads to problems when connected to other components with `fso:{supplies,returns}FluidTo` and trying to follow individual fluid loops. Having the connection `fso:suppliesFluidTo` through a component would suggest that the same fluid flows through the component, while actually a heat exchanger has two sides that do not mix but only transfer heat.

A solution to this is to model the heat exchanger as two components: a primary and secondary side, that do not have a fluid connection but only a heat connection. This could be done marking the heat exchanger both `fso:System` and `fso:EnergyConversionDevice`, connecting to the two sides with `fso:hasComponent`. But this feels like a missed opportunity in the sense that a heat exchanger is not just a "logical collection" of components like a `fso:System` but rather a physical component in itself that can be decomposed into separate sides.

As such, I think adding `fso:hasSubComponent` and the inverse `fso:isSubComponentOf` could be useful for these use cases. With these, the heat exchanger and its two sides can be of class `fso:EnergyConversionDevice` and there is no need to use the `fso:hasComponent` of system decomposition.

This could look something like this:

```
<heatExchanger> a fso:EnergyConversionDevice ;
    fso:hasSubComponent <primarySide>, <secondarySide> .

<primarySide> a fso:EnergyConversionDevice ;
    fso:transfersHeatTo <secondarySide> .
<secondarySide> a fso:EnergyConversionDevice ;
    fso:suppliesFluidTo <downstream> .

<downstream> fso:returnsFluidTo <secondarySide> .
<upstream> fso:suppliesFluidTo <primarySide> .
<primarySide> fso:returnsFluidTo <upstream> .
```

A similar case is true for heating and cooling coils in air handling units. Modeling them as single components that have both air and water flowing through them makes following the fluids more complicated, if not impossible. I have previously modeled them as a `fso:Segment` where the air passes through, with a `fso:exchangesHeatWith` connection to a `fso:EnergyConversionDevice` where the water flows through.

For reference, in IFC 4 `IfcEnergyConversionDevice` has the subclasses [`IfcHeatExchanger`](https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/schema/ifchvacdomain/lexical/ifcheatexchanger.htm) and [`IfcCoil`](https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/schema/ifchvacdomain/lexical/ifccoil.htm) (among others), where ports are used to differentiate the different flow. I'm not how they're in general used to follow a specific fluid, but it is possible in theory.

**TL;DR: I suggest we add `fso:hasSubComponent` and `fso:isSubComponentOf` with domain and range of `fso:Component` for component decomposition.**
